### DISCUSSION TITLE
Additional embedded media on set pages such as video

### DISCUSSION THREAD

**Discussion by suspect15** [RetroAchievements/RAWeb#1273] [Ideas]
I threw together a movie style trailer for a set I recently released yesterday to practice some video editing and thought it might be interesting to be able to embed a video/trailer on a set's page near the other media (title screen, game footage).  I have very little video creating/experience and threw this together in a couple hours, so it isn't tough to make them.

 I posted a poll on discord and there were definitely some mixed opinions, but it was about 75% liked the idea of adding it to the set page and 25% didn't.  Several of the comments did say that it made them more interested in the set than they previously had been.

I see some see benefits and drawbacks that I'll list here.  I'm curious what the most invested folks' opinions are.  Here's a link to the demo I made: [https://www.youtube.com/watch?v=Wh2InD_YUnk](https://www.youtube.com/watch?v=Wh2InD_YUnk)

Pros:
- Some find it adds to the overall presentation of the set
- Can show players much more about a game in a short digestible format than screenshots can
- Can showcase some of the best/most attractive parts of the game from someone who knows it better than most (the set dev)
- May add to user's overall RA experience
- May get some players more hyped up/excited to play a set that has a cool trailer

Cons:
- May be clunky for mobile users (brought up by a discord user)
- Could be viewed as advertising sets or vying for players amongst devs
- Not all devs would have time or interest in doing something like this, especially given how many sets have already been released.  Standardization among set pages would suffer

Anyway, just spitballing to see if anyone likes this idea.

**Comment by Jamiras** [comment]
I'm not sure if it's exactly the same idea, but #737 proposes embedding gameplay videos.

**Comment by cIymax** [comment]
How about just adding a sidebar link for now?  Something like this:

![image](https://user-images.githubusercontent.com/94894395/237000325-37cbbd3e-df6e-4959-8e37-36f3110af1eb.png)

**Overview**
- This could encourage more devs to make trailers for their sets, thus enriching what the site has to offer our users.
- Dev privileges would be needed to add a set trailer link to a set page.
- For now, it won't be an embed, but at least a set trailer could finally be linked to a game page for users to reference and enjoy.  Embedding is more challenging from a page design and UX perspective, and the sidebar link lets us defer that till later while at least getting the backend ready.  This discussion can be left open until the embedding piece is resolved.
- At least initially, we can limit the links to Youtube videos only.  This can ease the need for moderation. 

**Considerations**
- [#1408](https://github.com/RetroAchievements/RAWeb/pull/1408) added a sidebar link for Guides, which addresses [#1051](https://github.com/RetroAchievements/RAWeb/discussions/1051).
- The Guide link is also shown in my screenshot so we can see how everything fits together.
- [#737](https://github.com/RetroAchievements/RAWeb/discussions/737) is related but can be distinct imo, because set highlights and gameplay highlights can focus on very different things, depending on the game and the intended theme of the set.
- [Link](https://discord.com/channels/310192285306454017/1105369231119368283) to companion topic on Discord.

**Reply by wescopeland** [reply]
"Set Trailer" sounds like an action ("I'm going to set a trailer") rather than metadata about the game or its set. 

Though there are a few places in the UI that refer to them as "sets" such as some of the widgets on the homepage, I am actually starting to believe this is a power user term that most casual users don't understand, and the inclination to label things this way is probably an anti-pattern. For example, https://github.com/RetroAchievements/RAWeb/pull/1521 changed "HIde completed sets" to "Hide completed games".

I am wondering if such a trailer is primary enough that it doesn't belong in the right-hand metadata area. I would imagine these are going to be exceptionally rare, even over a very long period of time, thus I think this solution will lead to them being missed by most players.

For achievement videos they are embedded onto the page itself:
https://retroachievements.org/achievement/307541

An achievements set trailer is unique enough content that I'm wondering if a better solution is to bury the 2 screenshots on the game page somehow and show the trailer instead.

**Reply by cIymax** [reply]
- The label can be made more descriptive, e.g., "Watch Trailer for this [Achievement] Set."
- Long-term, I do think it belongs in the main area nearer the game screenshots and whatnot.  If it ends up there, the link should be removed from the sidebar.
- This is just intended as a stopgap solution at least until we have a promising layout in place for an embed in the main area.  That said, if we can come up with such a layout in the near term, I'm also fine with skipping the sidebar stopgap and going straight to the embed.
- I do agree that the sidebar link can be easily missable by users.  We have the same problem as RA Guides, but perhaps worse, as set trailers may be even rarer than RA Guides, which are already rare.
- We have less space to work with on set pages than we have on achievement pages, so I'd imagine it'd be harder to come up with a layout to add embeds to the former.

**Reply by wescopeland** [reply]
I'm not convinced that space is an issue. Current video embeds are ~350px in height. For most games, the images occupy ~250px in height. We already have code for embedding videos onto a page. If such a trailer is attached, we can bury the 2 screenshots under a collapse or something and show the video instead. You are right that this UX probably needs a good solution, so maybe the sidebar link is a compromise stopgap for now.

Another thing we will need to consider is, similar to guides, devs will be inclined to link to anything and everything in this field, especially to content that RA does not necessarily control. https://github.com/RetroAchievements/RAWeb/pull/1553 addresses this with RAGuides, however policing/enforcement seems more challenging here.

**Reply by wescopeland** [reply]
> Another thing we will need to consider is, similar to guides, devs will be inclined to link to anything and everything in this field, especially to content that RA does not necessarily control. https://github.com/RetroAchievements/RAWeb/pull/1553 addresses this with RAGuides, however policing/enforcement seems more challenging here.

A good solution here might be similar to how RAPatches has a centralized GitHub repo, we have a centralized "RATrailers" (I am spitballing) YouTube channel where trailers can be uploaded and preserved.

**Reply by cIymax** [reply]
- While there's some implicit trust in devs, another solution is to require manual approval by a mod team to elevate a dev's submitted video link to approved status to be shown on a game page.
- If trailers stay this rare, this hopefully won't impose too much of a burden on the mod team.  There's certainly pros and cons to each approach.
- As it stands, devs can already link a video to an achievement page.  But I can see that more moderation may be needed for a game page due to its higher visibility compared to achievement pages.
- I think I remember seeing that juniors will also be able to link a video to an achievement page (in Unofficial, I think).  If so, maybe this could be a reason why we can consider letting full devs (but not juniors) freely link a video to game pages without moderation after all.
- Or, we can just go back to the sidebar link again partly so we won't have to worry about moderating such an obscure sidebar link anyhow 😸
- Another idea: a roulette like the one we have on the main page, that cycles between showing the trailer embed versus the game screenshots.

**Reply by wescopeland** [reply]
> As it stands, devs can already link a video to an achievement page. But I can see that more moderation may be needed for a game page due to its higher visibility compared to achievement pages.

This is my thinking as well. If this is indeed a stopgap solution, I think for a non-stopgap to be developed afterwards the policing part has to unfortunately still be solved with the initial release of the feature.

Mod approval could be a viable approach, one downside though is the content is only implicitly controlled by RAdmin. If the user pulls their channel down (or YouTube's heavy hand strikes their account due to other content on their channel), nothing will stop the link from being broken onsite. This also solves a potentially murky monetization problem if the video trailers are indeed going to be primary metadata for the game. If the videos are linked from any channel, RA can't control how that content is (or isn't**) monetized. With a centralized YouTube account dedicated to these, it can.

It is true that Juniors have historically not been able to link videos on achievement pages. Resolved in https://github.com/RetroAchievements/RAWeb/pull/1526 😅
